src/Types/NatRatio.purs

```diff
@@ -0,0 +1,78 @@
+-- | A module for constructing well-defined ratios of natural numbers (with
+-- | non-zero denominator), potentially via cancellation of negative sign.
+module Types.NatRatio
+  ( NatRatio
+  , denominator
+  , denominatorAsNat
+  , fromBigInts
+  , fromNaturals
+  , fromRational
+  , numerator
+  , numeratorAsNat
+  , toRational
+  ) where
+
+import Prelude
+import Data.BigInt (BigInt)
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Ratio ((%), denominator, numerator) as R
+import Types.Natural (Natural, fromBigInt', toBigInt)
+import Types.Rational (Rational)
+
+-- | `NatRatio` is a newtype over `Rational` with smart constructors to create a
+-- | `NatRatio` safely. Therefore, the data constructor is not exported.
+newtype NatRatio = NatRatio Rational
```

Comment:
Can we remove Rational and do

```
newtype Rational = Rational (Ratio BigInt)
```

?

Rational isn't going to be used much, right?

Reply:
I was told that it's used a lot, I'd need to investigate how the Plutus Rational is implemented and probably replicate that, regardless of zero denoms unfortunately. Also, I believe they have a type alias too.

Reply:
Sorry, I meant the old `type Rational = Ratio BigInt` isn't going to be used because the intention is to use the safe newtype.

So let's remove the type synonym in order not to occupy a good name.

Reply:
So `NatRatio` is meant to be different to `Rational`, not a replacement, `NatRatio` is just the "non-negative version" so we do need both. Although, perhaps a `Newtype` does make sense for `Rational`. This issue is the default behaviour of Purescript `Ratio` may different to Haskell's `Ratio`, in which case a newtype would make sense to replicate Haskell's behaviour, and therefore Pltuus.